### DISCUSSION TITLE
Question about new cursor position behavior of `nvim_buf_set_text`

### DISCUSSION THREAD

**Discussion by MunifTanjim** [neovim/neovim#25883] [Q&A]
I was debugging an issue (https://github.com/zbirenbaum/copilot.lua/issues/216) and stumbled upon this: https://github.com/neovim/neovim/pull/24901

```lua
-- add two numbers
-- fu

vim.api.nvim_win_set_cursor(0, { 2, 4 })
vim.api.nvim_feedkeys("a", "n", false)
vim.lsp.util.apply_text_edits({
  {
    range = {
      ["end"] = {
        character = 5,
        line = 1,
      },
      start = {
        character = 0,
        line = 1,
      },
    },
    newText = "-- function add(a, b)\n--    return a + b\n-- end",
  },
}, 0, "utf-16")
```

Put that into a file and then run `:luafile %`.

Where should the cursor be after that?

I'm getting this:

```lua
-- add two numbers
-- fu|nction add(a, b)
--    return a + b
-- end
```

but I was expecting this:

```lua
-- add two numbers
-- function add(a, b)
--    return a + b
-- end|
```

Is tis expected behavior? Or is this a bug?

Related Links:
- https://github.com/zbirenbaum/copilot.lua/issues/216
- https://github.com/nvim-neo-tree/neo-tree.nvim/issues/1025

**Comment by miversen33** [comment]
It might be worth raising this as an issue to potentially get more eyes on it
